### PR TITLE
adding is_volume flag to filters

### PR DESCRIPTION
**Pull request by shanaxel42**
Fixes: #760

### CODE REVIEW

File: starfish/image/_filter/richardson_lucy_deconvolution.py [hunk 1]
```diff
@@ -6,15 +6,23 @@
...
+    gaussian_kernel,
+    preserve_float_range
+)
 
 
 class DeconvolvePSF(FilterAlgorithmBase):
 
     def __init__(
-            self, num_iter: int, sigma: Number, clip: bool=True) -> None:
+            self,
```
Comment (ttung) [inline starfish/image/_filter/richardson_lucy_deconvolution.py]: btw, if this fits on one line, you can go ahead and make it all on one line.  my general order of preferences:
1. fit on one line.
2. fit on two lines.
3. one param per line.

File: starfish/image/_filter/zero_by_channel_magnitude.py [hunk 2]
```diff
@@ -28,6 +28,7 @@ def __init__(self, thresh: int, normalize: bool) -> None:
         """
         self.thresh = thresh
         self.normalize = normalize
+        self.is_volume = is_volume
```
Comment (ttung) [inline starfish/image/_filter/zero_by_channel_magnitude.py]: doesn't look like anyone uses this...?

File: starfish/image/_filter/richardson_lucy_deconvolution.py [hunk 3]
```diff
@@ -27,6 +35,7 @@ def __init__(
         clip : bool (default = False)
             if True, pixel values below -1 and above 1 are clipped for skimage pipeline
             compatibility
+            :param is_volume:
```
Comment (ttung) [inline starfish/image/_filter/richardson_lucy_deconvolution.py]: this is not numpy doc style.

Comment (shanaxel42) [reply starfish/image/_filter/richardson_lucy_deconvolution.py]: oh man I think pycharm auto did this

File: starfish/image/_filter/zero_by_channel_magnitude.py (hunk 2 above)
Comment (joshmoore) [reply starfish/image/_filter/zero_by_channel_magnitude.py]: @shanaxel42 : I wasn't sure either which is why I passed the buck. 😉

Comment (ttung) [reply starfish/image/_filter/zero_by_channel_magnitude.py]: My intuition is that we should not include this as it implies it does something it does not.  I think there are filters that do not accept this format.

Second choice: raise ValueError if is_volume is True.

Comment (shanaxel42) [reply starfish/image/_filter/zero_by_channel_magnitude.py]: Updated to not include the flag here

**Pull request merged**